zaza/model.py

```diff
@@ -1740,3 +1741,81 @@ def attach_resource(application, resource_name, resource_path):
     cmd = ["juju", "attach-resource", "-m", juju_model,
            application, "{}={}".format(resource_name, resource_path)]
     subprocess.check_call(cmd)
+
+
+async def run_on_machine(machine, command, model_name=None, timeout=None):
+    """Juju run on unit.
+
+    This function uses a spawned process to run the `juju run` command rather
+    that a native libjuju call as libjuju hasn't implemented `juju.Machine.run`
+    yet.
+
+    :param model_name: Name of model unit is in
+    :type model_name: str
+    :param unit_name: Name of unit to match
+    :type unit: str
+    :param command: Command to execute
+    :type command: str
+    :param timeout: How long in seconds to wait for command to complete
+    :type timeout: int
+    :returns: action.data['results'] {'Code': '', 'Stderr': '', 'Stdout': ''}
+    :rtype: dict
+    """
+    cmd = ['juju', 'run', '--machine={}'.format(machine)]
+    if model_name:
+        cmd.append('--model={}'.format(model_name))
+    if timeout:
+        cmd.append('--timeout={}'.format(timeout))
+    cmd.append(command)
+    logging.info("About to call '{}'".format(cmd))
+    await generic_utils.check_call(cmd)
+
+
+sync_run_on_machine = sync_wrapper(run_on_machine)
```

Comment:
We have tended not to use the 'sync_' prefix for the non-async versions but I don't feel that strongly about it